src/Sql/Sql/Database Backup/Cmdlet/AzureSqlDatabaseBackupShortTermRetentionPolicyCmdletBase.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.Azure.Commands.Sql.Backup.Services;
+using Microsoft.Azure.Commands.Sql.Common;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Azure.Commands.Sql.Backup.Model;
+using Microsoft.Azure.Commands.Sql.Database.Model;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.Sql.Backup.Cmdlet
+{
+    public abstract class AzureSqlDatabaseBackupShortTermRetentionPolicyCmdletBase : AzureSqlCmdletBase<IEnumerable<AzureSqlDatabaseBackupShortTermRetentionPolicyModel>, AzureSqlDatabaseBackupAdapter>
+    {
+        /// <summary>
+        /// The expected number of segments in a short term retention policy resource id.
+        /// </summary>
+        private const int BackupShortTermRetentionPolicyResourceIdSegmentsLength = 12;
+
+        /// <summary>
+        /// Parameter set with ResourceGroup name, Server name and Database name.
+        /// </summary>
+        protected const string PolicyByResourceServerDatabaseSet = "PolicyByResourceServerDatabaseSet";
+
+        /// <summary>
+        /// Parameter set for using a Database Input Object.
+        /// </summary>
+        private const string PolicyByInputObjectSet = "PolicyByInputObjectSet";
+
+        /// <summary>
+        /// Parameter set for using a resource Id.
+        /// </summary>
+        private const string PolicyByResourceIdSet = "PolicyByResourceIdSet";
+
+        /// <summary>
+        /// Gets or sets the Database object to get the policy for.
+        /// </summary>
+        [Parameter(
+            ParameterSetName = PolicyByInputObjectSet,
+            Mandatory = true,
+            ValueFromPipeline = true,
+            HelpMessage = "The database object to get the policy for.")]
+        [ValidateNotNullOrEmpty]
+        public AzureSqlDatabaseModel AzureSqlDatabase { get; set; }
```

Comment:
@dealaus nit: I'd rename this parameter as `AzureSqlDatabaseObject` (and add an alias to `AzureSqlDatabase`) to let users know to provide the object representation of the database and not a name